backend/src/controllers/webhookController.ts

```
@@ -0,0 +1,96 @@
+                        .from('subscriptions')
+                        .upsert(upsertData, { onConflict: 'stripe_subscription_id' });
+                    if (upsertError) throw upsertError;
+                    console.log('[Ctrl:Webhook] Upsert successful for:', subscription.id);
+                    break;
+
+                case 'customer.subscription.deleted':
+                    console.log('[Ctrl:Webhook] Marking subscription canceled:', subscription.id);
+                    const { error: deleteError } = await supabase
+                        .from('subscriptions')
+                        .update({ status: 'canceled' as SubscriptionStatus, updated_at: new Date().toISOString() })
+                        .eq('stripe_subscription_id', subscription.id);
```

Biome (1.9.4)

[error] Other switch clauses can erroneously access this declaration.
Wrap the declaration in a block to restrict its access to the switch clause.

The declaration is defined in this switch clause:

Unsafe fix: Wrap the declaration in a block.


(lint/correctness/noSwitchDeclarations)